core/src/test/scala/com/banno/kafka/ProcessingAndCommittingSpec.scala

```diff
@@ -360,4 +386,182 @@ class ProcessingAndCommittingSpec extends CatsEffectSuite with KafkaSpec {
     }
   }
 
+  test("keepalive stream reads committed offsets on restart") {
+    producerResource.use { producer =>
+      for {
+        groupId <- IO(genGroupId)
+        topic <- createTestTopic[IO]()
+        p = new TopicPartition(topic, 0)
+        ps = Set(p)
+        keepAlive = 1.second
+        records = (0 to 9).toList.map(v => new ProducerRecord(topic, v, v))
+        count = records.size.toLong
+        _ <- producer.sendAsyncBatch(records)
+        cpause <- consumerResource(GroupId(groupId)).use { consumer =>
+          for {
+            _ <- consumer.subscribe(topic)
+            c0 <- consumer.partitionQueries.committed(ps)
+            pac = consumer.processingAndCommitting(
+              pollTimeout = 100.millis,
+              maxRecordCount = 1,
+              maxElapsedTime = Long.MaxValue.nanos,
+              keepAlive,
+            )(_.value.pure[IO])
+            results <- pac.take(count / 2).compile.toList
+            c1 <- consumer.partitionQueries.committed(ps)
+          } yield {
+            assert(c0.isEmpty)
+            assertEquals(results, (0 to 4).toList)
+            c1
+          }
+        }
+        crestart <- consumerResource(GroupId(groupId)).use { consumer =>
+          for {
+            _ <- consumer.subscribe(topic)
+            c0 <- consumer.partitionQueries.committed(ps)
+            pac = consumer.processingAndCommitting(
+              pollTimeout = 100.millis,
+              maxRecordCount = 1,
+              maxElapsedTime = Long.MaxValue.nanos,
+              keepAlive,
+            )(_.value.pure[IO])
+            results <- pac.take(count).interruptAfter(100.millis).compile.toList
+            c1 <- consumer.partitionQueries.committed(ps)
+          } yield {
+            assertEquals(results, (5 to 9).toList)
+            assertEquals(c0(p).offset, count / 2)
+            assertEquals(c1(p).offset, count)
+            c0
+          }
+        }
+      } yield {
+        assertEquals(cpause(p).offset, count / 2)
+        assertEquals(crestart(p).offset, count / 2)
+      }
+    }
+  }
+
+  // skip slow tests with,
+  //   core/testOnly com.banno.kafka.ProcessingAndCommittingSpec -- --exclude-tags=slow
+  test("offsets expire with no keepalive (slow)".tag(slow)) {
+    producerResource.use { producer =>
+      for {
+        _ <- IO.println("slow test >= 75 seconds")
+        groupId <- IO(genGroupId)
+        topic <- createTestTopic[IO]()
+        p = new TopicPartition(topic, 0)
+        values = (0 to 9).toList
+        records = values.map(v => new ProducerRecord[Int, Int](topic, v, v))
+        pstream = Stream.emits[IO, ProducerRecord[Int, Int]](
+          records.take(5)
+        ) ++ Stream.sleep_[IO](75.seconds) ++ Stream
```

Comment:
If this duration is changed at any point, we would want the `println` above to change as well. Is it worth putting this into a variable so that they always stay in sync?